core/dbt/linker.py

```diff
@@ -153,6 +155,7 @@ def mark_done(self, node_id):
             self.graph.remove_node(node_id)
             self._find_new_additions()
             self.inner.task_done()
+            self.some_task_done.notify_all()
```

Comment:
Do we need to change the context manager here to use `with self.some_task_done` instead of `with self.lock`? I assume it's equivalent since `some_task_done` just owns the lock, but that's what `queue.Queue` does.

Reply:
That's a great advise 🥇  . I was thinking that it was at least safe and working. I know it should be more consistent with `self.lock`. I'll be working about it :wink:. Thanks a lot for suggestion.

Reply:
Ok. You're right that it's same to use `self.lock` as the Condition is actually based upon it. I've changed that to be more confident with other GraphQueue methods as suggested.